docs/blog/2019-10-17-how-gatsby-can-power-your-site-offline-even-without-javascript/index.md

```diff
@@ -0,0 +1,31 @@
+date: 2019-10-17
+author: David Bailey
+excerpt: >
+  With Gatsby's offline plugin, network troubles are a thing of the past.
+tags:
+  - plugins
+  - performance
+  - accessibility
+---
+
+You know the feeling: you saw a cool website or read an interesting article the other day, and you'd like to check back on it, or maybe you’re with some people that you’d like to share it with. But what’s this? You don’t have any phone signal, and the wi-fi connection is spotty at best – if there is one at all! You attempt to open the page, but try as you might, the loading spinner just keeps on spinning...
+
+With `gatsby-plugin-offline`, that's a thing of the past. Install the plugin in an existing Gatsby site using NPM or Yarn, then add it to your `gatsby-config.js`, and your site will instantly gain offline support, without any additional configuration needed. ([Click here for full instructions on how to do this.](https://www.gatsbyjs.org/docs/add-offline-support-with-a-service-worker/)) Now, whenever someone visits a page on your site, that page will be available without an internet connection, along with any other pages which have been preloaded in the background.
+
+Recently, `gatsby-plugin-offline` version 3 was released, which added some exciting new features such as the ability to pre-cache a customized set of pages, as well as making some under-the-hood changes, the most notable being a major version upgrade of Workbox, the library which powers the core functionality of `gatsby-plugin-offline`. However, arguably the most interesting change is that the plugin will now keep working even if you disable JavaScript on a website which uses it.
+
+How does this work? Service workers and page-level JavaScript run in different contexts and don't have direct access to each other's context, although they can communicate with each other using a built-in message API - or even via data contained within a request, since requests can be intercepted by the service worker. More on that point later.
+
+`gatsby-plugin-offline` has a wrapper API which makes use of the message API, enabling the page-level JavaScript to transfer data to/from the service worker - for example, this can be used to specify which JavaScript and CSS resources are essential for a particular page. Its service worker also serves a lightweight "offline shell" for pages which are available from the cache (determined by checking for each of the essential resources) instead of the full pre-rendered HTML.
+
+When you (or someone accessing your website) disable JavaScript for a particular domain, contrary to what you might expect, any service workers installed on that domain keep running. This means the plugin will continue to intercept page requests and serve a shallow offline shell if it determines that all the resources are available - even though JavaScript is required to display the content of the page in this case (since the offline shell doesn't contain pre-rendered HTML).
```

Comment:
```suggestion
When you (or someone accessing your website) disables JavaScript for a particular domain, contrary to what you might expect, any service workers installed on that domain keep running. This means the plugin will continue to intercept page requests and serve a shallow offline shell if it determines that all the resources are available - even though JavaScript is required to display the content of the page in this case (since the offline shell doesn't contain pre-rendered HTML).
```

Reply:
Here I conjugated "disable" in the you-form since "someone accessing" is in brackets - not sure what's the best approach here